metaserver/src/main/scala/scala/meta/languageserver/SymbolIndexer.scala

```diff
@@ -66,7 +68,32 @@ class SymbolIndexer(
   }
 
   private def definition(symbol: Symbol): Option[Position.Range] =
-    Option(definitions.get(symbol))
+    Option(definitions.get(symbol)).map {
+      case Position.Range(input @ Input.VirtualFile(path, _), start, end)
+          if path.contains("jar") =>
+        Position.Range(createFileInWorkspaceTarget(input), start, end)
+      case pos => pos
+    }
+
+  // Writes the contents from in-memory source file to a file in the target/source/*
+  // directory of the workspace. vscode has support for TextDocumentContentProvider
+  // which can provide hooks to open readonly views for custom uri schemes:
+  // https://code.visualstudio.com/docs/extensionAPI/vscode-api#TextDocumentContentProvider
+  // However, that is a vscode only solution and we'd like this work for all
+  // text editors. Therefore, we write instead the file contents to disk in order to
+  // return a file: uri.
+  private def createFileInWorkspaceTarget(
```

Comment:
Just as a comparison, the Java LSP jumps to a `.class` file containing the source.
I'm not sure I understand the implementation, but here's the relevant bit https://github.com/eclipse/eclipse.jdt.ls/blob/2127f12fb3700dd783b75d973e4f2ecc134b9ddf/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/NavigateToDefinitionHandler.java#L52